src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+        and puts them in a new column.
+
+        Valid FERC license numbers are:
+        - Greater than 2.
+        - Usually accompanied by key phrases such as "license", "no.", "ferc", or
+        "project".
+        - Don't contain phrases like "page", "pg", "$", "wind", "units"
+        - Don't fall within the range of a valid year, defined as: 1900-2050
+        - Are categorized as either hydro or NA.
+
+        This function also fills "other" fuel types with hydro for all these plants.
+        """
+        logger.info(f"{self.table_id.value}: Extracting FERC license from plant name")
+        # Extract all numbers greater than 2 digits from plant_name_ferc1 and put them
+        # in a new column as integers.
+        out_df = df.assign(
+            license_id_ferc1=lambda x: (
+                x.plant_name_ferc1.str.extract(r"(\d{3,})")
+                .astype("float")
+                .astype("Int64")
+            ),
+        )
+        # Define what makes a good license
+        obvious_license = out_df.plant_name_ferc1.str.contains(
+            r"no\.|license|ferc|project", regex=True
+        )
+        not_license = out_df.plant_name_ferc1.str.contains(
+            r"page|pg|\$|wind|solar|nuclear|nonutility|units|surrendered", regex=True
+        )
+        exceptions = out_df.plant_name_ferc1.str.contains(
+            r"tomahawk|otter rapids|wausau|alexander|hooksett|north umpqua", regex=True
+        )
+        year_vs_num = (out_df["license_id_ferc1"] > 1900) & (
+            out_df["license_id_ferc1"] < 2050
+        )
+        not_hydro = ~out_df["plant_type"].isin(["hydro", np.nan, None]) | ~out_df[
+            "fuel_type"
+        ].isin(["hydro", "other"])
+        # Replace all the non-license numbers with NA
+        out_df.loc[
+            (not_hydro & ~obvious_license)
+            | not_license
+            | (year_vs_num & ~obvious_license & ~exceptions),
+            "license_id_ferc1",
+        ] = np.nan
+        # Fill fuel type with hydro
+        out_df.loc[
+            out_df["license_id_ferc1"].notna() & (out_df["fuel_type"] == "other"),
+            "fuel_type",
+        ] = "hydro"
+
+        return out_df
+
+    def label_row_type(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Label rows as headers, notes, or totals."""
+        # Define header qualifications
+        header_strings = [
+            "hydro",
+            "hyrdo",
+            "internal",
+            "wind",
+            "solar",
+            "gas",
+            "diesel",
+            "diesal",
+            "steam",
+            "other",
+            "combustion",
+            "combustine",
+            "fuel cell",
+            "hydraulic",
+            "waste",
+            "landfill",
+            "photovoltaic",
+            "nuclear",
+            "oil",
+            "renewable",
+            "facilities",
+            "combined cycle",
```

Comment:
Ideally we would connect this to one of the params, but we need to figure that out...

Reply:
i think adding these as params would be be a great idea!

Reply:
Still undecided whether these should be their own param....might help de-clutter the other outputs too...

Reply:
For now, I'm leaving them as is.